torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -275,6 +338,104 @@ struct KinetoThreadLocalState : public ProfilerThreadLocalState {
         generateForwardBackwardLink(kineto_event, fwd_bwd_link_id, activity, tidSeq2activity);
       }
     }
+
+    addPythonEvents();
+  }
+
+  void addPythonEvents() {
+    auto py_events = python_tracer::get_events();
+    for (const auto& e : py_events) {
+      TORCH_INTERNAL_ASSERT(
+        !e->thread_id_,
+        "Profiler expects only single threaded Python tracing.");
+    }
+
+    // The remainder of this function merges the Python and Kineto event
```

Comment:
My thinking was that rather than try to write a second set of "bailout" logic that would have to be kept in sync, I just make `python_tracer::get_events()` return an empty vector, and then the cases of (1) we're in pure C++ land, (2) the tracer was not enabled, and (3) no events were recorded all get to use the simple and natural bailout return here. (Making an empty vector should be very cheap.) WDYT?